.circleci/config.yml

```diff
@@ -267,6 +270,25 @@ jobs:
           name: Persist performance snapshot as pipeline artifact
           path: performance-snapshot.json
           destination: performance-snapshot.json
+      - when:
+          # don't run on PRs
+          condition:
+            not:
+              matches:
+                # "^pull/\d+" is not valid YAML
+                # "^pull/\\d+" matches neither 'pull/1' nor 'main'
+                # Note that we want to include 'pull/1', 'pull/1/head' and ''pull/1/merge'
+                pattern: '^pull/.+$'
+                value: << pipeline.git.branch >>
+          steps:
+            # persist performance snapshot on S3
+            - aws-s3/copy:
+                arguments: --content-type application/json
+                aws-access-key-id: AWS_ACCESS_KEY_ID_ARTIFACTS
+                aws-region: AWS_REGION_ARTIFACTS
+                aws-secret-access-key: AWS_SECRET_ACCESS_KEY_ARTIFACTS
+                from: performance-snapshot.json
+                to: s3://mui-org-ci/artifacts/$CIRCLE_BRANCH/$CIRCLE_SHA1/
```

Comment:
I would suggest we isolate the different repositories. This is what I see on https://s3.console.aws.amazon.com/s3/buckets/mui-org-ci?region=eu-central-1&prefix=artifacts/pull/29876/9f74e56b5e340db2004ea566c128da687ab0a509/&showversions=false

<img width="815" alt="Screenshot 2022-02-07 at 16 39 53" src="https://user-images.githubusercontent.com/3165635/152820660-65188893-fb50-48e0-83dc-6af473ee7f20.png">

Either maybe with a file prefix

```suggestion
                from: mui-x-performance-snapshot.json
                to: s3://mui-org-ci/artifacts/$CIRCLE_BRANCH/$CIRCLE_SHA1/
```

or with a repository prefix:

```suggestion
                from: performance-snapshot.json
                to: s3://mui-org-ci/artifacts/mui-x/$CIRCLE_BRANCH/$CIRCLE_SHA1/
```

Reply:
IMO, hierarchically, the repo prefix makes most sense.
```
from: size-snapshot.json
to: s3://mui-org-ci/artifacts/$CIRCLE_PROJECT_REPONAME/$CIRCLE_BRANCH/$CIRCLE_SHA1/
```
I would suggest to do the same in the material-ui repo, and maintain the same file name (`size-snapshot`) so that we don't have to special case the contributor-dashboard table